app/controllers/api/skater_profiles_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Api::SkaterProfilesController < ApplicationController
+  def show
+    profile = find_profile(current_skater.id)
+
+    render json: profile
+  end
+
+  def create
+    profile = SkaterProfile.create!(create_params)
+
+    render json: profile
+  end
+
+  def update
+    profile = find_profile(current_skater.id)
+    profile.update_attributes!(update_params)
+
+    render json: profile
+  end
+
+  private
+
+  def find_profile(current_skater_id)
+    SkaterProfile.find_by(skater_id: current_skater_id)
+  end
+
+  def create_params
+    params.require(:profile).permit(:skater_id, :first_name, :last_name, :avatar_file_path)
```

Comment:
If you let the user edit the skater_id column, they could create a new profile and set it to belong to any skater_id, or they could update their profile to belong to a different skater_id.

Remove skater_id from the params.permit() arguments in both the create_params and update_params. Though, you still want create_params to return a hash that has a skater_id key (use current_skater.id as the value).